test/schemas/tests-schema-0.0.1.json

```diff
@@ -0,0 +1,445 @@
+              "$ref": "#/$defs/anyExp"
+            }
+          }
+        },
+        {
+          "properties": {
+            "tests": {
+              "type": "array",
+              "items": {
+                "$ref": "#/$defs/anyExp"
+              }
+            }
+          }
+        }
+      ]
+    }
+  ],
+  "$defs": {
+    "defaultTestProperties": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "testType": {
+          "$ref": "#/$defs/testType"
+        },
+        "locale": {
+          "$ref": "#/$defs/locale"
+        },
+        "src": {
+          "$ref": "#/$defs/src"
+        },
+        "params": {
+          "$ref": "#/$defs/params"
+        },
+        "exp": {
+          "$ref": "#/$defs/exp"
+        },
+        "expCleanSrc": {
+          "$ref": "#/$defs/expCleanSrc"
+        },
+        "expParts": {
+          "$ref": "#/$defs/expParts"
+        },
+        "expErrors": {
+          "$ref": "#/$defs/expErrors"
+        }
+      }
+    },
+    "test": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "description": {
+          "type": "string",
+          "description": "Information about the test."
+        },
+        "testType": {
+          "$ref": "#/$defs/testType"
+        },
+        "locale": {
+          "$ref": "#/$defs/locale"
+        },
+        "src": {
+          "$ref": "#/$defs/src"
+        },
+        "params": {
+          "$ref": "#/$defs/params"
+        },
+        "exp": {
+          "$ref": "#/$defs/exp"
+        },
+        "expCleanSrc": {
+          "$ref": "#/$defs/expCleanSrc"
+        },
+        "expParts": {
+          "$ref": "#/$defs/expParts"
+        },
+        "expErrors": {
+          "$ref": "#/$defs/expErrors"
+        },
+        "only": {
+          "type": "boolean",
+          "description": "Normally not set. A flag to use during development to only run one or more specific tests."
+        }
+      }
+    },
+    "testType": {
+      "description": "The category of test.",
+      "enum": [
+        "syntax",
+        "function"
+      ]
+    },
+    "locale": {
+      "description": "The locale to use for formatting.",
+      "type": "string"
+    },
+    "src": {
+      "description": "The MF2 syntax source.",
+      "$comment": "Array values should be concatenated into a single string. They are only included to help readability in the test JSON.",
+      "type": [
+        "string",
+        "array"
+      ],
+      "items": {
+        "type": "string"
+      }
+    },
+    "params": {
+      "description": "Parameters to pass in to the formatter for resolving external variables.",
+      "type": "array",
+      "items": {
+        "$ref": "#/$defs/var"
+      }
+    },
+    "var": {
+      "type": "object",
+      "oneOf": [
+        {
+          "additionalProperties": false,
+          "required": [
+            "value"
+          ],
```

Comment:
Shouldn't `"name"` also be required? Same goes for the other `"var"` variant.

Reply:
This was originally done to allow `params` to take positional args. e.g.
```
"params": [
  { "value": "first" },
  { "value": "second" },
]
```
I don't think we need to do this though - `params` can just be an `object`.

Reply:
Yes, you're right.
`name` was originally optional to support positional args. I don't think we need this though.